### PR TITLE
ui/map:  singleton navigation requests

### PR DESCRIPTION
**Pull request by deanlee**
There are multiple instances of `MapSettings` in home and map, each with its own `RequestRepeater` to query from commaapi for navigation requests.  these duplicate requests result in wasted bandwidth and resources, and may potentially lead to synchronization issues.

### CODE REVIEW

File: selfdrive/ui/qt/maps/map_settings.h [hunk 1]
```diff
@@ -23,6 +23,21 @@ const QString NAV_FAVORITE_LABEL_WORK = "work";
 class NavDestination;
 class DestinationWidget;
 
+class NavigationRequest : public QObject {
+  Q_OBJECT
+
+public:
+  NavigationRequest(QObject *parent);
```
Comment (incognitojam) [inline selfdrive/ui/qt/maps/map_settings.h:30]: should we make the constructor non-public to avoid doing this again accidentally?

**Comment by incognitojam** [comment]
The destination list is empty onroad, first starting after this commit :confused:

**Comment by incognitojam** [comment]
It does seem related as I can't reproduce the issue on the prior commit (d34138e2759e8067e9d7d005957e29a23684c71c)

**Comment by deanlee** [comment]
I'm checking to see what the possible problem is. It's unlikely to be this pr's problem.

**Comment by deanlee** [comment]
@incognitojam it's Indeed a related issue.  `NavigationRequest` only reads the locations from `ApiCache_NavDestinations` on initialization and then `emit locationsUpdated`. this results in only the first `mapSetting` window can gets an immediate update. the second `mapsetting` can only be updated when `NavigationRequest` gets the list from the server, which can take up to 30 seconds.

**Pull request merged**